cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/graph/statements/expressions/LoopTest.kt

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph.statements.expressions
+
+import de.fraunhofer.aisec.cpg.GraphExamples
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.callsByName
+import de.fraunhofer.aisec.cpg.graph.functions
+import de.fraunhofer.aisec.cpg.graph.statements
+import de.fraunhofer.aisec.cpg.graph.statements.*
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class LoopTest {
+
+    @Test
+    fun testWhileStatement() {
+        val whileTest = GraphExamples.getWhileWithElseAndBreak()
+        val func = whileTest.functions["someRecord.func"]
+        assertNotNull(func)
+        val whileStmt = whileTest.statements.filterIsInstance<WhileStatement>().firstOrNull()
+        assertNotNull(whileStmt)
+        assertTrue { func.body.statements.contains(whileStmt) }
+        val breakStmt = whileStmt.statements.filterIsInstance<BreakStatement>().firstOrNull()
+        val elseCall = whileTest.callsByName("elseCall").getOrNull(0)
+        val postWhile = whileTest.callsByName("postWhile").getOrNull(0)
+        assertNotNull(breakStmt)
+        assertNotNull(elseCall)
+        assertNotNull(postWhile)
+        assertTrue(whileStmt.astChildren.all { whileStmt.toString().contains(it.toString()) })
+    }
+
+    @Test
+    fun testDoStatement() {
+        val doTest = GraphExamples.getDoWithElseAndBreak()
+        val func = doTest.functions["someRecord.func"]
+        assertNotNull(func)
+        val doStmt = doTest.statements.filterIsInstance<DoStatement>().firstOrNull()
+        assertNotNull(doStmt)
+        assertTrue { func.body.statements.contains(doStmt) }
+        val breakStmt = doStmt.statements.filterIsInstance<BreakStatement>().firstOrNull()
+        val elseCall = doTest.callsByName("elseCall").getOrNull(0)
+        val postWhile = doTest.callsByName("postDo").getOrNull(0)
+        assertNotNull(breakStmt)
+        assertNotNull(elseCall)
+        assertNotNull(postWhile)
+        assertTrue(doStmt.astChildren.all { doStmt.toString().contains(it.toString()) })
+    }
+
+    @Test
+    fun testForStatement() {
+        val forTest = GraphExamples.getForWithElseAndBreak()
+        val func = forTest.functions["someRecord.func"]
+        assertNotNull(func)
+        val forStmt = forTest.statements.filterIsInstance<ForStatement>().firstOrNull()
```

Comment:
minor nitpicking. Usually we try to always have blocks of variables + their assert and then newline, then next var again.